File: ButtRydeServer/Models/DepartmentViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace AASC.Partner.API.Models
{
    public class DepartmentViewModel
    {
        public string Id { get; set; }

        public String Name { get; set; }

        public string CompanyId { get; set; }

        public virtual CompanyViewModel Company { get; set; }

        public string ParentDepartmentId { get; set; }

        public virtual DepartmentDisplayViewModel ParentDepartment { get; set; }

        public string DepartmentHeadEmployeeId { get; set; }

        public virtual EmployeeViewModel DepartmentHead { get; set; }

        public DateTime? CreatedDate { get; set; }

        public string CreatedById { get; set; }

        public virtual DisplayUserBindingModel CreatedBy { get; set; }

        public virtual ICollection<EmployeeListViewModel> Employees { get; set; }

        public virtual ICollection<DepartmentViewModel> ChildrenDepartments { get; set; }
    }

    public class DepartmentListViewModel
    {
        public string Id { get; set; }

        public String Name { get; set; }

        public string CompanyId { get; set; }

        public string CompanyName { get; set; }

        public string ParentDepartmentId { get; set; }

        public string ParentDepartmentName { get; set; }

        public string DepartmentHeadEmployeeId { get; set; }

        public string DepartmentHeadUserName { get; set; }        

        public DateTime? CreatedDate { get; set; }

        public string CreatedById { get; set; }

        public string CreatedByUserName { get; set; }

        public virtual ICollection<EmployeeViewModel> Employees { get; set; }

        public virtual ICollection<DepartmentListViewModel> ChildrenDepartments { get; set; }
    }

    public class DepartmentDisplayViewModel
    {
        //public Guid? Id { get; set; }
        public string Id { get; set; }

        public string Name { get; set; }
    }
}

File: ButtRydeClient/ButtRydeClient/app/viewModels/departmentViewModel.js
'use strict';
/*
    Obsolete: 05/10/2016
    Use departmentListViewModel instead
*/
app.factory('departmentViewModel', function () {
    return kendo.data.Model.define({
        id: 'id',
        fields: {
            id: { type: 'string', editable: false, nullable: false },
            name: { type: 'string', editable: true, nullable: false },
            companyId: { type: 'string', editable: false, nullable: false },
            company: { 
                type: 'companyViewModel',
                editable: false,
                defaultValue: {
                    id: '',
                    name: ''
                }
            },
            parentDepartmentId: { type: 'string', editable: false, nullable: true },
            parentDepartment: {                 
                //id: { type: 'string' },
                //name: { type: 'string' }
                type: 'departmentDisplayViewModel',
                editable: false,                
                defaultValue: {
                    id: '',
                    name: '',
                }
            },

            departmentHeadEmployeeId: { type: 'string', editable: false, nullable: true },
            departmentHead: {
                type: 'employeeViewModel',
                editable: false,
                defaultValue: {
                    id: '',
                    applicationUser: {
                        id: '',
                        username: '',
                        firstName: '',
                        lastName: '',
                        email: ''
                    }
                }
            },
            createdById: { type: 'string', editable: false, nullable: false, defaultValue: '00000000-0000-0000-0000-000000000000' },
            createdDate: {
                type: 'date',
                editable: false,
                nullable: true
            },
            createdBy: {
                type: 'displayUserBindingModel',
                editable: false,
                defaultValue: {
                    id: '',
                    email: '',
                    username: '',
                    firstName: 'firstName',
                    lastName: 'lastName'
                }
            },
            childrenDepartments: []
        }
    });
});